File: Dart/test/findIntersectionValues_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:leetcode/findIntersectionValues.dart';

void main() {
  test("test findIntersectionValues", () {
    var input0 = [
      [2, 3, 2],
      [4, 3, 2, 3, 1],
      [3, 4, 2, 3]
    ];
    var input1 = [
      [1, 2],
      [2, 2, 5, 2, 3, 6],
      [1, 5]
    ];
    var expected = [
      [2, 1],
      [3, 4],
      [0, 0]
    ];
    final sln = Solution();
    for (var i = 0; i < input0.length; i++) {
      var output = sln.findIntersectionValues(input0[i], input1[i]);
      expect(output, expected[i]);
    }
  });
}

File: Typescript/test/findIntersectionValues.test.ts
import { expect, test } from '@jest/globals';
import * as solution from '../src/findIntersectionValues';

test('Example 1', () => {
	let input1 = [2, 3, 2];
	let input2 = [1, 2];
	let expected = [2, 1];
	expect(solution.findIntersectionValues(input1, input2)).toEqual(expected);
});

test('Example 2', () => {
	let input1 = [4, 3, 2, 3, 1];
	let input2 = [2, 2, 5, 2, 3, 6];
	let expected = [3, 4];
	expect(solution.findIntersectionValues(input1, input2)).toEqual(expected);
});

test('Example 3', () => {
	let input1 = [3, 4, 2, 3];
	let input2 = [1, 5];
	let expected = [0, 0];
	expect(solution.findIntersectionValues(input1, input2)).toEqual(expected);
});
